rosbag2_py/src/rosbag2_py/format_bag_metadata.cpp

```diff
@@ -124,17 +126,105 @@ void format_topics_with_type(
       info_stream << std::endl;
     };
 
-  print_topic_info(topics[0]);
   size_t number_of_topics = topics.size();
-  for (size_t j = 1; j < number_of_topics; ++j) {
+  size_t i = 0;
+  // Find first topic which isn't service event topic
+  while (i < number_of_topics &&
+    rosbag2_cpp::is_service_event_topic(
+      topics[i].topic_metadata.name,
+      topics[i].topic_metadata.type))
+  {
+    i++;
+  }
+
+  if (i == number_of_topics) {
+    return;
+  }
+
+  print_topic_info(topics[i]);
+  for (size_t j = ++i; j < number_of_topics; ++j) {
+    if (rosbag2_cpp::is_service_event_topic(
+        topics[j].topic_metadata.name, topics[j].topic_metadata.type))
+    {
+      continue;
+    }
     indent(info_stream, indentation_spaces);
     print_topic_info(topics[j]);
   }
 }
 
+struct ServiceMetadata
+{
+  std::string name;
+  std::string type;
+  std::string serialization_format;
+};
+
+struct ServiceInformation
+{
+  ServiceMetadata service_metadata;
+  size_t event_message_count;
```

Comment:
```suggestion
  size_t event_message_count = 0;
```
Add default initialization for the `event_message_count` to avoid access to the garbage value.